### ISSUE TITLE
Unable to turn on or off the orca screenreader on arctica-greeter lightdm greeter using <alt>+<super>+<s>

### ISSUE THREAD

**Issue by ricmarques** [ArcticaProject/arctica-greeter#38]
Bug originally reported in Launchpad, for the "arctica-greeter package" present in Ubuntu MATE 20.10 and 21.04, by Ubuntu MATE's Team Member [N0rbert](https://github.com/N0rbert), on 9th April 2021 (2021-04-09). Yesterday (15th October 2022), I ("[ricmarques](https://github.com/ricmarques)" = "Ricardo Dias Marques") have checked that this bug is still present / still happens in Ubuntu MATE 22.04.1 LTS ("Jammy Jellyfish") which uses "arctica-greeter" ("LightDM Arctica Greeter") version "0.99.1.5-2nmu3":

Bug #1923253 “Unable to turn on or off the orca screenreader on arctica-greeter lightdm greeter using &lt;alt&gt;+&lt;super&gt;+&lt;s&gt;” : Bugs : arctica-greeter package : Ubuntu
https://bugs.launchpad.net/ubuntu/+source/arctica-greeter/+bug/1923253

Also yesterday (15th October 2022), I've added a comment - https://bugs.launchpad.net/ubuntu/+source/arctica-greeter/+bug/1923253/comments/5 - in that bug, with the following text:

> This problem is also happening in my laptop computer running Ubuntu MATE 22.04.1 LTS ("Jammy Jellyfish"): when I'm in the login screen, if I press the "Alt + Windows key + S" key combination, the (Orca) Screen Reader does NOT start. The screen reader only starts if I go to the accessibility icon menu at the top of the screen and then click on the "Screen Reader" option. After I log in, the "Alt + Windows key + S" key combination works correctly in Ubuntu MATE to turn on and off the Orca screen reader.

This bug is being discussed, for instance, in the following Discussion in the "Ubuntu MATE Community" Discussion Forum:

CRITICAL! Still unable to startup orca at the login screen - Support &amp; Help Requests - Ubuntu MATE Community
https://ubuntu-mate.community/t/critical-still-unable-to-startup-orca-at-the-login-screen/25938

**Comment by Ionic** [comment]
Unfortunately, I cannot reproduce this issue with our current code.

I can reproduce the issue with an Ubuntu MATE 22.04 live DVD, so it definitely has been happening. I then tried the Ubuntu MATE 23.04 live DVD and it is working as intended there, with 0.99.2.0.

I have no idea what might have caused it in older versions. The code that setups the menu bar entry and the keyboard shortcut hasn't been changed in ages.

Usually, I'd close this bug report now, since there's nothing upstream can do to fix something that isn't broken any longer.

However, I realize that it might be worthwhile to fix the issue in the Ubuntu LTS version and this needs a backport of whatever fixed it. I'll try bisecting the code tomorrow to hopefully find out what has coincidentally made it work so that this change can be backported to the old version in Ubuntu 22.04.

**Comment by Ionic** [comment]
Wow, this was surprisingly difficult to debug.

The issue is caused by `mate-settings-daemon` capturing global shortcuts, of which <kbd>Super</kbd>-<kbd>Meta</kbd>-<kbd>S</kbd> is one (naturally toggling the screen reader).

Thus, the greeter never had a chance to actually see the keypresses and the accelerator (keyboard shortcut for a menu entry) never triggered.

Now, we certainly don't want MSD to capture any shortcuts (or do other weird things), which is why a lot of MSD modules are disabled in Arctica-Greeter.

Unfortunately, this was buggy up to 0.99.2.0, which means that it disabled nothing.

@sunweaver please backport 0199f9ac4ddb2f70ff94e651b51a77e81d9383ca and e09a615c972150f6eaff336257c7d042845ad59e to the Ubuntu package for jammy. That should fix the issue.

I'll also close this issue report because there really isn't anything wrong in our current version.

**Comment by sunweaver** [comment]
@Ionic @ricmarques Unfortunately, I am not a Ubuntu developer and cannot update packages that have landed in Ubuntu releases. I can only improve the future. In Ubuntu, the required patches are available starting with 23.04.

**Comment by Ionic** [comment]
Hm, what about @flexiondotorg? He might be interested in fixing the greeter for 22.04, has NMU'd the package recently and backporting the patches is easy, since they apply as-is.
